gio/src/subclass/list_model.rs

```diff
@@ -0,0 +1,92 @@
+use crate::ListModel;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, IsA, ObjectExt};
+
+pub trait ListModelImpl: ObjectImpl {
+    fn get_item_type(&self, list_model: &Self::Type) -> glib::Type;
+    fn get_n_items(&self, list_model: &Self::Type) -> u32;
+    fn get_item(&self, list_model: &Self::Type, position: u32) -> Option<glib::Object>;
+}
+
+unsafe impl<T: ListModelImpl> IsImplementable<T> for ListModel
+where
+    <T as ObjectSubclass>::Type: IsA<glib::Object>,
```

Comment:
Instead of all these `where` clauses you could also just use the unsafe qdata API from `gobject_sys` btw